app/src/main/java/ru/prometeydev/movie/data/MoviesRepository.kt

```diff
@@ -0,0 +1,81 @@
+package ru.prometeydev.movie.data
+
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.withContext
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.json.Json
+import ru.prometeydev.movie.domain.JsonActor
+import ru.prometeydev.movie.domain.JsonGenre
+import ru.prometeydev.movie.domain.JsonMovie
+
+class MoviesRepository(private val dispatcher: CoroutineDispatcher) {
```

Comment:
Just wonder why do you inject dispatcher?

Reply:
Probably can not injectt, I will make a private field in MoviesRepository